packages/@tailwindcss-upgrade/src/migrate-js-config.ts

```diff
@@ -232,3 +251,9 @@ function onlyUsesAllowedTopLevelKeys(theme: ThemeConfig): boolean {
   }
   return true
 }
+
+function keyframesToCss(keyframes: Record<string, unknown>): string {
+  let ast: AstNode[] = []
+  applyKeyframesToAst(ast, { theme: { keyframes } })
+  return toCss(ast).trim() + '\n'
```

Comment:
That's a sneaky way of handling this. I love it!

Reply:
for real this is a pretty sweet way of doing it 💯